### PR TITLE
Draw live histogram for blink latency test

### PR DESCRIPTION
**Pull request by sahiljain**
Draws a histogram in realtime, showing the latency distribution for black-to-white and white-to-black separately.

# Future Work
+ Configure `BIN_WIDTH` in settings
+ Once test is finished, expand to fullscreen? Add option to minimize/maximize
hist-1

### CODE REVIEW

File: android/WALT/app/src/main/java/org/chromium/latency/walt/ScreenResponseFragment.java [hunk 1]
```diff
@@ -110,9 +114,14 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
         blackBox.setMovementMethod(new ScrollingMovementMethod());
-        chartLayout = view.findViewById(R.id.chart_layout);
+        brightnessChartLayout = view.findViewById(R.id.brightness_chart_layout);
+        latencyChartLayout = view.findViewById(R.id.latency_chart_layout);
         view.findViewById(R.id.button_close_chart).setOnClickListener(this);
-        chart = (LineChart) view.findViewById(R.id.chart);
+        view.findViewById(R.id.button_close_latency_chart).setOnClickListener(this);
+        brightnessChart = (LineChart) view.findViewById(R.id.chart);
+        latencyChart = (HistogramChart) view.findViewById(R.id.latency_chart);
+        latencyChart.setLabel(W2B_INDEX, "White-to-black Latency");
```
Comment (kamrik) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/ScreenResponseFragment.java]: Let's drop the "Latency", but add M=median when the test is done. (or in some other place on the chart).

File: android/WALT/app/src/main/java/org/chromium/latency/walt/HistogramChart.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+        XAxis xAxis = getXAxis();
+        xAxis.setCenterAxisLabels(true);
+        xAxis.setGranularityEnabled(true);
+        xAxis.setGranularity(1);
+        xAxis.setValueFormatter(new IAxisValueFormatter() {
+            DecimalFormat df = new DecimalFormat("#.##");
+
+            @Override
+            public String getFormattedValue(float value, AxisBase axis) {
```
Comment (kamrik) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/HistogramChart.java]: Let's switch to lables and ticks between the bars at N x BIN_WIDTH

Comment (sahiljain) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/HistogramChart.java]: Done!

File: android/WALT/app/src/main/java/org/chromium/latency/walt/HistogramChart.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+        if (isRawDataEmpty()) {
+            min = value;
+            max = value;
+        } else {
+            if (value < min) min = value;
+            if (value > max) max = value;
+        }
+
+        rawData.get(dataSetIndex).add(value);
+        minBin = (int) (Math.floor(min/BIN_WIDTH)*BIN_WIDTH);
```
Comment (sahiljain) [inline android/WALT/app/src/main/java/org/chromium/latency/walt/HistogramChart.java]: Sounds good, I've updated it to be `floor`, and now the labels appear on the boundaries.

**Comment by kamrik** [comment]
Looks good. Merging.

**Pull request merged**